### PR TITLE
Refactored distance to capital stability

### PR DESCRIPTION
**Pull request by Grummel7**
Fixed the happiness calculation for distance from capital or regional administration.
It should use the minimum of all supply connected candidates.
Also used to named refs to make it easier to maintain.

### CODE REVIEW

File: default/scripting/species/common/happiness.macros [hunk 1]
```diff
@@ -69,84 +69,62 @@ COMMON_HAPPINESS_EFFECTS
...
+JUMPS_TO_CAPITAL_OR_REGAD
+'''
+		Statistic Min
+		    // Replace JumpsBetween by JumpsBetweenByEmpireSupplyConnections once it is implemented
+            value = JumpsBetween
+            object = Source.ID
+            object = LocalCandidate.ID
+            condition = And [
+                Or [Capital Building name = "BLD_REGIONAL_ADMIN"]
+                OwnedBy empire = Source.Owner
```
Comment (geoffthemedio) [inline default/scripting/species/common/happiness.macros:90]: why not use `SUPPLY_CONNECTED_TO_CAPITAL_OR_REGAD` here?

Comment (Grummel7) [reply default/scripting/species/common/happiness.macros:90]: Which part do you want to replace, the lines above your comment or the one line below?
The condition must identify the Capital and all RAs, but only if they are connected to the Source, i.e. the planet to be affected by the happiness modification. The Capital itself is of course always supply connected to the Capital.

File: default/scripting/species/common/happiness.macros [hunk 2]
```diff
@@ -69,84 +69,62 @@ COMMON_HAPPINESS_EFFECTS
             ]
 '''
 
+SUPPLY_CONNECTED_TO_CAPITAL_OR_REGAD
+'''
+		// this should ensure that there is a capital / admin for JUMPS_TO_CAPITAL_OR_REGAD
```
Comment (geoffthemedio) [inline default/scripting/species/common/happiness.macros]: replace tabs with spaces everywhere

File: default/scripting/species/common/happiness.macros (hunk 1 above)
Comment (geoffthemedio) [reply default/scripting/species/common/happiness.macros:90]: Actually, neither. I was misreading the structure of the condition in the macro above due to linebreaks, in part due to the tab characters causing a lot of indentation.

Comment (Grummel7) [reply default/scripting/species/common/happiness.macros:90]: Oh yeah, fun with tabs. At work we use an SVN hook to stop any check-in containing a tab.
The default settings for text editors in eclipse apparently create them. I changed the settings and updated the commit.

**Pull request merged**